### ISSUE TITLE
Problems loggin data with sweep 

### ISSUE THREAD

**Issue by Pharaun85** [wandb/wandb#1529]
Hello
I'm working with wandb in the optimization of a network and since I updated to the last version I found some problems with the sweep that I didn't find before. 
My code runs like this:
```
for train_index, test_index in kf.split(dataset_train):
     run = wandb.init(project="3d-deep", group=group_id, job_type="training", reinit=True)
     wandb.config.update(args)
     F1_result = train(args, model, run, optimizer, dataloader_train, dataloader_val) #Train loop with run.log items
     F1_record.append(F1_result)
     run.finish()

run = wandb.init(project="3d-deep", group=group_id, job_type="eval")
F1_record = np.array(F1_record)
F1_mean = np.mean(F1_record)
F1_max = np.max(F1_record)
run.log({"Val/F1_mean": F1_mean,
              "Val/F1_max": F1_max}, step=0)
```

The thing is that if I launch the code without a sweep works fine and all the folds are logged as a different run but grouped by, and the F1_max and F1_mean are logged too. However, if I create a sweep from the web console of my project the runs don't group each other and overwrite the previous ones in the graph and the F1_mean and F1_max values don't appear anywhere. That didn't happen in previous versions where the method to use was wandb.join() instead of run.finish() . Also, as a minor problem, the steps in my graphs are not rounded numbers despite the step that I'm passing to the log function is the epoch, that is an integer. 

Please, can you help me with this?

**Comment by tyomhak** [comment]
Hi @Pharaun85 , thank you for flagging this. We're looking into it. What was the last version of wandb that this code worked with? 
Also, can you share what OS you're running the sweep on?

cc @raubitsj

**Comment by Pharaun85** [comment]
Thanks for your response. I was using version 0.10.4 and now I use 0.10.11
The OS version is Ubuntu 18.04.4 
Also, I don't know if it matters, but the metric that I use in the sweep is F1_mean. 
Thanks

**Comment by ariG23498** [comment]
Hey @Pharaun85 
I have seemed to reproduce your errors. I have made myself a simple implementation of the code that you shared, focussing mainly on the grouping problem that you have mentioned.

The link to the wandb project: https://wandb.ai/repro/sweep_group

## Train.py
```python
import wandb
defaults = dict(seed=42, filter_size=3)
F1_record = []
for g_id in ["a","b","c","d",]:
    run = wandb.init(entity="repro",
    project="sweep_group",
    group=g_id,
    job_type="training",
    config=defaults,
    )
    F1_result = 2*wandb.config["filter_size"] - 0.01*wandb.config["seed"]
    F1_record.append(F1_result)
    run.finish()

run = wandb.init(entity="repro",
                project="sweep_group",
                group="a",
                job_type="eval")
F1_record = F1_record
F1_mean = sum(F1_record)/len(F1_record)
F1_max = max(F1_record)
run.log({"Val/F1_mean": F1_mean,
              "Val/F1_max": F1_max}, step=0)
```

## Sweep.yml
```yaml
project: sweep_group
entity: repro
program: train.py
method: bayes
metric:
  goal: minimize
  name: output
parameters:
  seed:
    distribution: int_uniform
    max: 100
    min: 42
  filter_size:
    distribution: int_uniform
    max: 7
    min: 3
```

My observation is similar to yours:
- Without `sweep` the python script `train.py` runs as usual.
- While **sweeping** the groups seem to be overwritten of somehow get removed.

Please confirm with this minimal implementation. Let me know if I missed something here :smile:

**Comment by Pharaun85** [comment]
Thank you very much for answering. 
I think that the only difference I find is that in my case the group_id has the same name both in the training loop and in the final evaluation execution to group all the kfold of a single run together with its average metric. This name only changes when the sweep launches another execution of the train.py since it is the system time. 

```
args = parser.parse_args()
group_id = strftime("%H:%M:%S", gmtime())

print(args)
# Launch execution
main(args)
```

**Comment by pkuderov** [comment]
I have exactly the same problem. Any progress on it or a workaround?

**Comment by maltemosbach** [comment]
@vanpelt I am having the same problem. Do you have an update on how to make this work?

**Comment by vanpelt** [comment]
@maltemosbach this is a rather long thread.  Can you please share the details about what you're seeing?

**Comment by vanpelt** [comment]
@maltemosbach ahh, I see.  When running in a sweep the agent sets WANDB_RUN_ID which is causing each of the init methods to log to the same run.  We should handle this better in the SDK, but today you should be able to add the following after your first `wandb.finish()` call to make the eval run appear as a separate run:

```python
del os.environ['WANDB_RUN_ID']
```

**Comment by vanpelt** [comment]
For anyone else coming to this thread, there's a great [StackOverflow thread ](https://stackoverflow.com/questions/63469762/weightsbiases-sweep-keras-k-fold-validation) covering more advanced sweep use cases.  In the [example linked](https://github.com/wandb/examples/blob/master/examples/wandb-sweeps/sweeps-cross-validation/train-cross-validation.py) in that thread we have a more general purpose env reset function that can be used in cases where `wandb.init` is called multiple times for a single trial.

```python
def reset_wandb_env():
    exclude = {
        "WANDB_PROJECT",
        "WANDB_ENTITY",
        "WANDB_API_KEY",
    }
    for k, v in os.environ.items():
        if k.startswith("WANDB_") and k not in exclude:
            del os.environ[k]
```

**Comment by sydholl** [comment]
We are closing this issue due to inactivity. If you feel that it merits additional review, please provide a reproduction and add a comment explaining why. Thank you.
